### PR TITLE
fix export command

### PR DESCRIPTION
**Pull request by moonso**
This PR fixes some problems with the export cases command.

## datetime
The date time objects was not serialised in a correct way. Previously they where dumped in the json string as `datetime.datetime(2019, 6, 3, 10, 13, 47, 762000)` which is not possible for a json decoder to read. Now they are serialised as standardised strings on the format `"2020-05-06T09:26:04.348000"`

## status=solved

When running command `scout export cases --status solved` the events was not collected which made it impossible to fetch cases together with the `--within-days` parameter

## non-existing causatives

When querying `get_causatives` with a case_id for a case without causatives there was a syntax error

## bson.ObjectID

These where also serialised in the wrong way before. Previously they where dumped to json like `ObjectId('5e3185fd124f63addb79384b')` which can not be parsed, but now the string is used.

**Expected outcome**:
Export cases command should behave as expected

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by moonso** [comment]
This has been tested on stage!

**Comment by northwestwitch** [comment]
Which command doesn't work as it should?

**Comment by moonso** [comment]
Don't merge just yet! Think I need to fix one more thing

**Comment by moonso** [comment]
> Which command doesn't work as it should?

`scout export cases --status solved`

**Comment by northwestwitch** [comment]
Tested the command: `scout --demo export cases --status solved --within-days 2` on master:
image

This branch: 🆗 
image

Tested the command: 
`scout setup demo`
`scout --demo export cases --causatives --case-id internal_id`, and it works. So does in the PR. 🤔 
image

**Review by northwestwitch — APPROVED** [review]
I have some suggestions for improvements. Don't forget the changelog!

File: scout/commands/export/export_handler.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+"""Code for handler that helps export bson to json"""
+
+import datetime
+import bson
+
+
+def my_handler(x):
+    if isinstance(x, datetime.datetime):
+        return x.isoformat()
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/commands/export/export_handler.py:9]: Yes

File: scout/commands/export/variant.py [hunk 2]
```diff
@@ -112,16 +114,22 @@ def verified(collaborator, test, outpath=None):
 @click.option("--case-id", help="Find causative variants for case")
 @json_option
 @with_appcontext
-def variants(collaborator, document_id, case_id, json):
+def variants(collaborator: str, document_id: str, case_id: str, json: bool):
     """Export causatives for a collaborator in .vcf format"""
     LOG.info("Running scout export variants")
-    adapter = store
+    adapter: MongoAdapter = store
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/commands/export/variant.py]: Removed

File: scout/commands/export/export_handler.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+"""Code for handler that helps export bson to json"""
+
+import datetime
+import bson
+
+
+def my_handler(x):
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/commands/export/export_handler.py]: Changed to `json_handler`

**Comment by moonso** [comment]
Thanks for the review @northwestwitch ! I have updated the branch according to the suggestions

File: scout/commands/export/variant.py [hunk 4]
```diff
@@ -2,12 +2,14 @@
 import click
 import logging
 import datetime
+import json as json_lib
 from flask.cli import with_appcontext
-from bson.json_util import dumps
 from xlsxwriter import Workbook
 
 from scout.export.variant import export_variants, export_verified_variants
+from scout.adapter.mongo import MongoAdapter
```
Comment (northwestwitch) [inline scout/commands/export/variant.py:10]:
```suggestion
```

**Review by northwestwitch — CHANGES_REQUESTED** [review]

File: scout/commands/export/variant.py (hunk 4 above)
Comment (northwestwitch) [reply scout/commands/export/variant.py:10]: Didn't want to add a negative review, just a suggestion, sorry!

Comment (northwestwitch) [reply scout/commands/export/variant.py:10]: looks like it's not passing a test now

**Review by northwestwitch — APPROVED** [review]
👍

**Pull request merged**